### ISSUE TITLE
Validation Peers

### ISSUE THREAD

**Issue by yesoer** [yesoer/peersdb#10]
depends on #9 

Validation peers should :

1. check for new blocks & if they are contribution blocks
2. try validating the files they point to (can get the file via the CID in the block)

Now the result needs to somehow reach other nodes. Possible solutions are outlined in the following sections.

## Approach 1 Distributed validation information

This approach tries to be simple and secure

3. Each peer keeps their own validation records
4. when someone wants to know 
    1. they retrieve all info (signed by the sender) via pubsub 
        1. announce their wish via topic : "validation" with message data : their id + the files cid
        2. receive votes via topic : their id + the files cid
    2. count votes themselves
    3. maybe persist the result ?

Pro :
- simple to implement
- secure : no one can manipulate
- old data will be validated aswell

Con :
- while a validation peer is offline their info will not be available to others
    - counter arg : we may assume that peers are online because that's how they profit 
    - counter approach : force all peers to validate to maximize input
- if 5.3 is implemented, we will have some redundancy in the system
    - counter arg : it's very small data

## Approach 2 Gathered validation information

This approach aims to achieve higher availability of validation info by gathering results before persisting them, somewhere where they are available to everyone. (this is obviously a pro to all these approaches)

### Sub-Approach 2.1 Third-Party voting

A trusted third party could be used for voting and retrieving results.

Pro :
- simple
- secure
- little overhead

Con :
- this would centralize part of the software 
- makes us dependable on said third party.

### Sub-Approach 2.2 Persist all votes

Gather signed votes and add them to the eventlog as-is for everyone to count and verify.

Con : 
- as many as possible need to work through the same validation in a set timeframe
    - this could be communicated : leader election by who first finds a new block and tells others to validate it
- requires persistence of the validators pubkeys so their votes authenticity can be verified later on
    - this might be happening anyways with the pubsub keypair, need to check 
- when peers add blocks we cannot keep them from adding rubbish
    - counter arg : this can always happen, as long as the resulting block would be obviously corrupted it's fine

### Sub-Approach 2.3 Blockchain

A blockchain with a consensus algorithm may be added. But that is way out of scope for now.

### Sub-Approach 2.4

One could also do more research on off-chain voting which might be possible to incorporate directly into our system but the time I have at hand for the necessary research isn't enough.
